temba/msgs/models.py

```diff
@@ -1421,9 +1459,25 @@ def substitute_variables(cls, text, context, contact=None, org=None, url_encode=
         return evaluate_template(text, context, url_encode, partial_vars)
 
     @classmethod
+    def get_outgoing_metadata(cls, metadata, message_context, contact, org, channel):
+        metadata = json.loads(metadata)
+        quick_replies = metadata.get('quick_replies', None)
+        if quick_replies:
+            for counter, reply in enumerate(quick_replies):
+                (text, errors) = Msg.substitute_variables(reply, message_context, contact=contact, org=org)
+                if text:
+                    if contact.is_test:
+                        quick_replies[counter] = text
+                    else:
+                        channel_type = Channel.get_type_from_code(channel.channel_type)
+                        quick_replies[counter] = text[:channel_type.quick_reply_text_size]
+
+        return json.dumps(metadata)
+
+    @classmethod
     def create_outgoing(cls, org, user, recipient, text, broadcast=None, channel=None, high_priority=False,
                         created_on=None, response_to=None, message_context=None, status=PENDING, insert_object=True,
-                        attachments=None, topup_id=None, msg_type=INBOX, connection=None):
+                        attachments=None, topup_id=None, msg_type=INBOX, connection=None, metadata=None):
```

Comment:
this arg should be called quick_replies rather than metadata since quick_replies is just one item we put we put in metadata